controllers/events/trigger_controller.go

```diff
@@ -0,0 +1,535 @@
+				}
+			}
+
+			if sub.DeadLetterSink != nil && !deadLetterSinks[sub.DeadLetterSink] {
+				deadLetterSinks[sub.DeadLetterSink] = true
+				componentName := fmt.Sprintf(TriggerSinkComponentNameTmpl, trigger.Name, sub.DeadLetterSink.Ref.Namespace, sub.DeadLetterSink.Ref.Name)
+				sc.DeadLetterSinkComponentName = componentName
+				if !totalSinks[sub.DeadLetterSink] {
+					totalSinks[sub.DeadLetterSink] = true
+					component := &componentsv1alpha1.Component{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      componentName,
+							Namespace: trigger.Namespace,
+						},
+					}
+					spec, err := newSinkComponentSpec(r.Client, r.Log, sub.DeadLetterSink.Ref)
+					if err != nil {
+						return err
+					}
+					sinkSpecList = append(sinkSpecList, spec)
+					component.Spec = *spec
+					components = append(components, component)
+				}
+			}
+
+			if sub.Topic != "" && !topics[sub.Topic] {
+				topics[sub.Topic] = true
+				sc.TopicName = sub.Topic
+				if !totalTopics[sub.Topic] {
+					totalTopics[sub.Topic] = true
+				}
+			}
+
+			if sub.DeadLetterTopic != "" && !deadLetterTopics[sub.DeadLetterTopic] {
+				deadLetterTopics[sub.DeadLetterTopic] = true
+				sc.DeadLetterTopicName = sub.DeadLetterTopic
+				if !totalTopics[sub.DeadLetterTopic] {
+					totalTopics[sub.DeadLetterTopic] = true
+				}
+			}
+
+			r.envs.Subscribers = append(r.envs.Subscribers, sc)
+		}
+		specListBytes, err := json.Marshal(sinkSpecList)
+		if err != nil {
+			return err
+		}
+
+		// Set the TriggerConfig.SinkSpecEncode which will reflect the specification content changes of dapr component back to the Trigger workload,
+		// informing it that it needs to rebuild.
+		r.envs.SinkSpecEncode = base64.StdEncoding.EncodeToString(specListBytes)
+
+	} else {
+		return errors.New("no subscriber found")
+	}
+
+	for _, component := range components {
+		c := component
+		if componentsPendingDelete[component.Name] {
+			componentsPendingDelete[component.Name] = false
+		}
+		if _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, c, mutateDaprComponent(r.Scheme, c, trigger)); err != nil {
+			return err
+		}
+		// Create the workload for Trigger.
+		if _, err := r.createOrUpdateTriggerWorkload(trigger, workloadsPendingDelete); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *TriggerReconciler) handleDeprecatedComponents(trigger *openfunctionevent.Trigger, cpd map[string]bool) error {
+	log := r.Log.WithName("handleDeprecatedComponents")
+	var component componentsv1alpha1.Component
+	for componentName, isDeprecated := range cpd {
+		if isDeprecated {
+			err := r.Get(r.ctx, types.NamespacedName{Namespace: trigger.Namespace, Name: componentName}, &component)
+			if err != nil {
+				if util.IsNotFound(err) {
+					continue
+
+				} else {
+					log.Error(err, "Failed to get deprecated component", "namespace", trigger.Namespace, "name", componentName)
+					return err
+				}
+			}
+			if err := r.Delete(r.ctx, &component); err != nil {
```

Comment:
I think we needn't to call get before delete, why not just delete the deprecated component and ignore the missing comp error by util.IsNotFound?

Reply:
Ignore this comments, the delete method require an object